crates/burn-compute/src/channel/base.rs

```diff
@@ -8,6 +11,12 @@ pub trait ComputeChannel<Server: ComputeServer>: Clone + core::fmt::Debug + Send
     /// Given a binding, returns owned resource as bytes
     fn read(&self, binding: Binding<Server>) -> Reader<Vec<u8>>;
 
+    /// Given a resource handle, return the storage resource.
+    fn get_resource(
+        &self,
+        binding: Binding<Server>,
+    ) -> <Server::Storage as ComputeStorage>::Resource;
```

Comment:
Get resource is better than read. Reading kind of assumes we actually do something with the memory, where get is simply returning the resource.